### PR TITLE
Switch to Kubernetes secret instead of users credentials

### PR DESCRIPTION
**Pull request by harshad16**
Switch to Kubernetes secret instead of users credentials
Closes: https://github.com/opendatahub-io/odh-dashboard/issues/1415

## Description

- switch the `cos_auth_type`: KUBERNETES_SECRET
https://elyra.readthedocs.io/en/latest/user_guide/runtime-conf.html#cloud-object-storage-authentication-type-cos-auth-type
- include `cos_secret` which reference the secret in the cluster with details of auth

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [ ] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Review by andrewballantyne — APPROVED** [review]
/hold

Let us test the image first before merging.

**Comment by harshad16** [comment]
Tested on a cluster:

Screenshot from 2023-06-21 13-30-55
The changes are showing up nicely in the generation.

**Comment by andrewballantyne** [comment]
/unhold

LGTM

**Pull request merged**
